zerver/views/streams.py

```diff
@@ -149,6 +148,15 @@ def update_stream_backend(
         new_name: Optional[Text]=REQ(validator=check_string, default=None),
 ) -> HttpResponse:
     (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id)
+    # Only realm admin and stream admin can update stream properties
+    # Realm admin can update unsubscribed public stream property but can not
+    # update unsubscribed private stream property.
+    if sub is None:
+        if not stream.invite_only and not user_profile.is_realm_admin:
```

Comment:
I think a local variable can help this be more clear, and I'm not sure the `and` is right here.  I think you mean:

~~~
is_public = not stream.invite_only
admin_modifying_public_stream = is_public and user_profile.is_realm_admin
if not admin_modifying_public_stream:
   # we have a non-subscriber with no special privileges, return an error
    return json_error(...)

Reply:
Yeah, it should be `or`.